62_CAD/MSDFs.cpp

```diff
@@ -0,0 +1,452 @@
+#include "DrawResourcesFiller.h"
+#include "MSDFs.h"
+#include "nbl/ext/TextRendering/TextRendering.h"
+
+// extents is the size of the MSDF that is generated (probably 32x32)
+// glyphExtents is the area of the "image" that msdf will consider (used as resizing, for fill patterns should be 8x8)
+DrawResourcesFiller::MsdfTextureUploadInfo generateMsdfForShape(std::vector<CPolyline>&& polylines, msdfgen::Shape glyph, uint32_t2 msdfExtents, uint32_t2 glyphExtents)
+{
+	uint32_t glyphW = msdfExtents.x;
+	uint32_t glyphH = msdfExtents.y;
+
+	if (polylines.empty())
+	{
+		return {
+			.cpuBuffer = nullptr,
+			.bufferOffset = 0ull,
+			.imageExtent = { glyphW, glyphH, 1 },
+			.polylines = std::move(polylines),
+		};
+	}
+
+	auto shapeBounds = glyph.getBounds();
+
+	msdfgen::edgeColoringSimple(glyph, 3.0); // TODO figure out what this is
+	msdfgen::Bitmap<float, 4> msdfMap(glyphW, glyphH);
+
+	float scaleX = (1.0 / float(glyphExtents.x)) * glyphW;
+	float scaleY = (1.0 / float(glyphExtents.y)) * glyphH;
+	msdfgen::generateMTSDF(msdfMap, glyph, MsdfPixelRange, { scaleX, scaleY }, msdfgen::Vector2(0.0, 0.0));
```

Comment:
why are you generating MTSDF (it has true sdf in alpha channel) but don't use it? you can just make the MSDF textures RGB and generateMSDF instead of MTSDF

Reply:
It was for debugging, easier to know what's wrong with the texture in render doc that way, but I'll change it over when everything is working

Reply:
Msdfs are harder to understand and read

Reply:
Or you can just use msdf.exe, generate the msdf and compare

Reply:
I can't use that with the hatch fills, since they're not exactly font files, but I'll change it over to msdf only when everything works